docs/developer/write-a-plugin.md

```diff
@@ -8,6 +8,20 @@ Whereas tasks used in jobs that define how the transformation will execute, what
 
 Optimus can be divided in two logical parts when we are thinking of a pluggable model, one is the **core** where everything happens which is common for all job/datastore, and the other part which could be variable and needs user specific definitions of how things should work which is a **plugin**.
 
+## Types of Plugins in Optimus
+At the moment mainly there are two types of plugins which optimus supports. These are : ***Hook*** and  ***Task***
+Before getting into the difference between two plugins ,we need to get familiar with [Jobs](../concepts/overview.md#Job).
+
+| Type                   | Hook                                                                                                                                          | Task                                                                                                                                                                                    |
+|------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Definition             | It is the operation that we preferably run before or after a Job.                                                                             | It is the single base transformation in a Job.                                                                                                                                          |
+| Fundamental Difference | It can have dependencies over other hooks within the job.                                                                                     | It can have dependencies over other jobs inside the repository.                                                                                                                         |
+| Configuration          | It has its own set of configs and share the same asset folder as the base job.                                                                | It has its own set of configs and may share only the dependencies with the other jobs in the repo.                                                                                      |
+| Examples               | 1. Transporter(BQ/GCS to Kafka)<br>2. Predator(Auditing & Profiling for BQ)<br>3. BeastLagChecker<br>4. Http Hooks<br>5. Tableau view updates | 1. BQ2BQTask - Transformation from BigQuery to BigQuery in SQL <br>2. SparkSQLTask - Transformation from BQ/GCS to BQ/GCS in SparkSQL <br>3. PySparkTask - Transformation using python. |
```

Comment:
Because these examples are not available to use at the moment, let's get rid of them.

Reply:
Should i remove all the examples or any specific?